libcaf_core/tests/legacy/flow/op/merge.cpp

```diff
@@ -86,6 +86,36 @@ SCENARIO("the merge operator combines inputs") {
       }
     }
   }
+  GIVEN("one fail observable with one successful observable") {
+    WHEN("merging them to a single observable") {
+      THEN("the observer aborts with error") {
+        using ivec = std::vector<int>;
+        auto snk = flow::make_auto_observer<int>();
+        make_observable()
+          .fail<int>(sec::runtime_error)
+          .merge(make_observable().repeat(22).take(223))
+          .subscribe(snk->as_observer());
+        ctx->run();
+        CHECK_EQ(snk->state, flow::observer_state::aborted);
+        CHECK_EQ(snk->err, sec::runtime_error);
+      }
+    }
+  }
+  GIVEN("two fail observables") {
+    WHEN("merging them to a single observable") {
+      THEN("the observer receives the error of first observable") {
+        using ivec = std::vector<int>;
+        auto snk = flow::make_auto_observer<int>();
+        make_observable()
+          .fail<int>(sec::runtime_error)
+          .merge(make_observable().fail<int>(sec::end_of_stream))
+          .subscribe(snk->as_observer());
+        ctx->run();
+        CHECK_EQ(snk->state, flow::observer_state::aborted);
+        CHECK_EQ(snk->err, sec::runtime_error);
```

Comment:
It's undefined which error will be observed. The current implementation may always yield `runtime_error`, but that's not required.

```suggestion
```